backend/routes/gallery/gallery.js

```diff
@@ -1,56 +1,56 @@
 import express from "express";
 import { validateCanvasData } from './validator.js';
-import { saveCanvasData, getAllCanvases } from "./canvas.js";
+import {saveCanvasData, getAllCanvases} from "./canvas.js";
 import { authenticate } from "../auth/authentication.js";
+import { paginatedResults } from "./paginatedResults.js";
 
 const router = express.Router();
 
 // POST Route to upload canvas Data
-router.post("/upload", authenticate, async (req, res) => {
-  try {
+router.post("/upload", authenticate, async(req, res) => {
+  try{
     const canvasData = req.body;
-
+    
     // validating the canvas data
     const validationError = validateCanvasData(canvasData);
-    if (validationError) {
+    if(validationError){
       return res.status(400).json({ success: false, error: validationError });
     }
 
     // Generating a unique filename for the image 
     // code here
 
     // Convert canvas to image
-    // code here (if we decide to add that. TBH the frontend is already capable of this)
+    // code here
 
     // Save canvas data to the database
     const result = await saveCanvasData(
       {
-        // manually destructured for security purposes
-        name: canvasData.name,
-        description: canvasData.description,
-        pixels: canvasData.pixels,
-        width: canvasData.width,
```

Comment:
Looks like you overwrote my changes with an old copy of the repo. Please fix.

Reply:
Sure

Reply:
Done